Cubical/HITs/EilenbergMacLane1/Properties.agda

```diff
@@ -0,0 +1,116 @@
+{-
+
+Eilenberg–Mac Lane type K(G, 1)
+
+-}
+
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+module Cubical.HITs.EilenbergMacLane1.Properties where
+
+open import Cubical.HITs.EilenbergMacLane1.Base
+
+open import Cubical.Core.Everything
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.HLevels
+
+open import Cubical.Data.Sigma
+
+open import Cubical.Algebra.Group.Base
+
+open import Cubical.HITs.PropositionalTruncation as PropTrunc using (∥_∥; ∣_∣; squash)
+open import Cubical.HITs.SetTruncation as SetTrunc using (∥_∥₂; ∣_∣₂; squash₂)
+
+private
+  variable
+    ℓG ℓ : Level
+
+module _ ((G , str) : Group {ℓG}) where
+
+  open GroupStr str
+
+  elimEq : {B : EM₁ (G , str) → Type ℓ}
+           (Bprop : (x : EM₁ (G , str)) → isProp (B x))
+           {x y : EM₁ (G , str)}
+           (eq : x ≡ y)
+           (bx : B x)
+           (by : B y) →
+           PathP (λ i → B (eq i)) bx by
+  elimEq {B = B} Bprop {x = x} =
+    J (λ y eq → ∀ bx by → PathP (λ i → B (eq i)) bx by) (λ bx by → Bprop x bx by)
+
+  elimProp : {B : EM₁ (G , str) → Type ℓ}
+             → ((x : EM₁ (G , str)) → isProp (B x))
+             → B embase
+             → (x : EM₁ (G , str))
+             → B x
+  elimProp Bprop b embase = b
+  elimProp Bprop b (emloop g i) = elimEq Bprop (emloop g) b b i
+  elimProp Bprop b (emcomp g h i j) =
+    isSet→SquareP (λ i j → isProp→isSet (Bprop (emcomp g h i j)))
+      (λ j → elimProp Bprop b (emloop g j))
+      (λ j → elimProp Bprop b (emloop (g + h) j))
+      (λ i → b)
+      (λ i → elimProp Bprop b (emloop h i))
+      i j
+  elimProp Bprop b (emsquash x y p q r s i j k) =
+    isOfHLevel→isOfHLevelDep 3 (λ x → isSet→isGroupoid (isProp→isSet (Bprop x)))
+    _ _ _ _ (λ j k → g (r j k)) (λ j k → g (s j k)) (emsquash x y p q r s) i j k
+    where
+      g = elimProp Bprop b
+
+  elimProp2 : {C : EM₁ (G , str) → EM₁ (G , str) → Type ℓ}
+            → ((x y : EM₁ (G , str)) → isProp (C x y))
+            → C embase embase
+            → (x y : EM₁ (G , str))
+            → C x y
+  elimProp2 Cprop c = elimProp (λ x → isPropΠ (λ y → Cprop x y))
+                               (elimProp (λ y → Cprop embase y) c)
+
+  elimSet : {B : EM₁ (G , str) → Type ℓ}
```

Comment:
Could eliminate some repetition here and in `GroupoidQuotients` by defining `elimSet` first and then using it to help define `elimProp`.

Reply:
Yes, that would be good. Can you give me a hint how to fill that hole? (I guess you just know what to do)

Reply:
Solved in meeting... (for EM-Spaces)

Reply:
Done